clconf/template.go

```diff
@@ -0,0 +1,208 @@
+package clconf
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+const setuid = 4
+const setgid = 2
+const sticky = 1
+
+// TemplateSettings are settings for ProcessTemplates.
+type TemplateSettings struct {
+	// Flatten determines if directory structure is preserved when scanning folders
+	// for templates.
+	Flatten bool
+	// KeepExistingPerms determines whether to use existing permissions on template files that are overwritten.
+	KeepExistingPerms bool
+	// Rm determines whether template files distinct from their target are deleted after processing.
+	Rm bool
+	// FileMode is a string of unix file permissions (e.g. 2755) that apply to templates. An empty string
+	// will copy permissions from the template itself.
+	FileMode string
+	// DirMode is the permission similar to FileMode but for new folders.
+	DirMode string
+	// Extension is the extension to use when searching folders. If missing all files will be used.
+	// The extension is stripped from the file name when templating.
+	Extension string
+}
+
+type pathWithRelative struct {
+	fullPath string
+	relPath  string
```

Comment:
More of a nitpick here, but _stutter_ is frowned upon, and `pathWithRelative.relPath` seems to do so.

Reply:
[Fixed](https://github.com/pastdev/clconf/pull/11/commits/dcc25840a6d3a22819b9a46795745c24915edec0)